### DISCUSSION TITLE
Any way to have a dev build side to side to the official release? (linux)

### DISCUSSION THREAD

**Discussion by WolfganP** [streamlink/streamlink#3737] [Q&A] [answered]
Hi, I'm trying to have a self contained dev/test build of streamlink from master to be installed side by side with the stable official distribution (via pip) and be available system wide to allow testing and better issues report & contributions back (ie having streamlink and streamlink-dev built and ready to quickly compare one vs another).

I have no problems doing it in an env as per https://streamlink.github.io/install.html#pypi-package-and-source-code but it's cumbersome to test the same command line back to back.

I tried packing of the pyc files ala youtube-dl
https://github.com/ytdl-org/youtube-dl/blob/dfbbe2902fc67f0f93ee47a8077c148055c67a9b/Makefile#L60
but it didn't work either (errors galore of not finding functions and packages).

I also tried a `python3 setup.py bdist_egg`, renamig the .egg and then use a custom forced launcher:
```
#!/usr/bin/python3
# -*- coding: utf-8 -*-
import re
import sys

sys.path = ['/myfolder/dist/streamlink-dev.egg'] + sys.path

from streamlink_cli.main import main
if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])
    sys.exit(main())
```

but it also fails to work properly.

I know it may be a simple and obvious thing, but I can't find the proper way of getting to it. Any tips?
Thanks in advance.

**Comment by bastimeyer** [comment, marked as answer]
> I have no problems doing it in an env as per [streamlink.github.io/install.html#pypi-package-and-source-code](https://streamlink.github.io/install.html#pypi-package-and-source-code) but it's cumbersome to test the same command line back to back.

Why is this cumbersome? Just create a new venv, activate it, clone the streamlink repo or any fork, check out your desired branch and install it in the venv via the `-e path/to/git/repo` parameter:
- https://github.com/streamlink/streamlink/blob/f06ae788cef28838c214069267b2226508c67f58/script/install-dependencies.sh#L10
- https://pip.pypa.io/en/stable/cli/pip_install/#install-editable

Then simply execute streamlink without having to activate the venv from the `$venv/bin/streamlink` path and update/change the git branch as much as you want (mind dependency changes though) and you can immediately run the recent changes.

> be available system wide

You can symlink it from `$HOME/.local/bin/streamlink-dev` or `/usr/bin/streamlink-dev`.

**Reply by WolfganP** [reply]
Will try it that way, thanks @bastimeyer !
